### DISCUSSION TITLE
Current reverse proxy documentation is garbage

### DISCUSSION THREAD

**Discussion by Kirutian** [nextcloud/all-in-one#3728] [Questions] [answered]
Trying to get students to follow the instructions on a Rocky Linux 9 environment. (Docker repo added via a centos method.)

None of me 2nd year students could do it. Then I tried. I couldn't do it following the instructions for an Apache proxy.

There is missing information and the documentation makes it seem complete. It's not.

Please remove it until its complete.

**Comment by Kirutian** [comment]
![image](https://github.com/nextcloud/all-in-one/assets/55530216/daf9befd-6611-4640-b299-ffc4ec83ed65)

**Comment by Kirutian** [comment]
This was the assigned run command we came up with in class:

sudo docker run \
--init \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 8080:8080 \
--env APACHE_PORT=11000 \
--env APACHE_IP_BINDING=0.0.0.0 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
--env NEXTCLOUD_MOUNT="/mnt/st0" \
--security-opt label:disable \
nextcloud/all-in-one:latest

**Comment by szaimen** [comment, marked as answer]
> Current reverse proxy documentation is garbage

Improvements to our documentation are very welcome!

Also see https://nextcloud.com/contribute/code-of-conduct/

And https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#6-how-to-debug-things

**Reply by Kirutian** [reply]
Ok. I will present to you how I read your documentation. Keep in mind I have extensive experience in computer programming and have been a Linux system administrator for 17 years.

**Comment by Kirutian** [comment]
Let's start with 

![image](https://github.com/nextcloud/all-in-one/assets/55530216/e034a135-c116-43c8-933c-fc5358d5b2d3)

Can you define what "chosen" APACHE_PORT is and how it is different from "$APACHE_PORT"

**Reply by Kirutian** [reply]
I'm assuming you're using '$' to denote a variable. However, this is confusing because you don't use variables anywhere else in your documentation. All good documentation has definitions and remains consistent.

**Reply by Zoey2936** [reply]
APACHE_PORT is the port you set in your docker run command/compose file as env, the "default" port used in the documentation is 11000, which is used in all reverse proxy exmaples

**Comment by Kirutian** [comment]
This is confusing and inconsistent:

![image](https://github.com/nextcloud/all-in-one/assets/55530216/d6ebab62-63ff-481b-871a-f7b78b5a9b48)

Are you saying that this documentation is ONLY for when the intention is for Nextcloud to be outside the masterconatiner?

**Reply by Zoey2936** [reply]
This point says exactly what is written there: the documentation is for nextcloud and not for the AIO panel

**Comment by Kirutian** [comment]
![image](https://github.com/nextcloud/all-in-one/assets/55530216/c55af820-183b-4ca2-8cc7-6594b646dfca)
#
Does this mean every place you see 
![image](https://github.com/nextcloud/all-in-one/assets/55530216/9561713a-eda8-4158-acfc-efca54614275)
you should treat it like a variable and input 11000 ?

**Comment by Kirutian** [comment]
What does this mean?
![image](https://github.com/nextcloud/all-in-one/assets/55530216/02d659b1-cf00-4bbe-8fcf-d2bf6a5a74b3)
